plugin.video.youtube/resources/lib/youtube_plugin/kodion/utils/player.py

```diff
@@ -1,102 +1,207 @@
 # -*- coding: utf-8 -*-
+"""
 
-import math
+    Copyright (C) 2018-2018 plugin.video.youtube
+
+    SPDX-License-Identifier: GPL-2.0-only
+    See LICENSES/GPL-2.0-only for more information.
+"""
+
+import re
 
 import xbmc
 
 
-class YouTubePlayer(xbmc.Player):
-    def __init__(self, *args, **kwargs):
-        self.context = kwargs.get('context')
-        self.ui = self.context.get_ui()
-        self.reset()
-        self.current_video_total_time = 0.0
-        self.current_played_time = 0.0
-
-    def reset(self):
-        properties = ['playing', 'post_play', 'license_url', 'license_token',
-                      'seek_time', 'play_count', 'playback_history', 'addon_id',
-                      'prompt_for_subtitles']
-        cleared = []
-        for prop in properties:
-            if self.ui.get_home_window_property(prop) is not None:
-                cleared.append(prop)
-                self.ui.clear_home_window_property(prop)
-        self.context.log_debug('Cleared home window properties: {properties}'.format(properties=str(cleared)))
-        self.current_video_total_time = 0.0
-        self.current_played_time = 0.0
-
-    def post_play(self):
-        is_playing = self.ui.get_home_window_property('playing')
-        post_play_command = self.ui.get_home_window_property('post_play')
-        use_playback_history = self.ui.get_home_window_property('playback_history') == 'true'
-
-        if is_playing is not None:
+def playback_monitor(provider, context, video_id, play_count=0, use_history=False,
+                     playback_stats=None, seek_time=None, refresh_only=False):
+    access_manager = context.get_access_manager()
+
+    player = xbmc.Player()
+    settings = context.get_settings()
+    ui = context.get_ui()
+
+    ui.set_home_window_property('video_id', video_id)
+
+    if playback_stats is None:
+        playback_stats = {}
+
+    play_count = str(play_count)
+
+    total_time = 0.0
+    current_time = 0.0
+    segment_start = 0.0
+    played_time = -1.0
+    percent_complete = 0
+
+    state = 'playing'
+    last_state = 'playing'
+
+    np_wait_time = 0.2
+    np_waited = 0.0
+    p_wait_time = 0.5
+    p_waited = 0.0
+
+    report_interval = 10.0
+    first_report = True
+
+    report_url = playback_stats.get('playback_url', '')
+
+    while not player.isPlaying() and not context.abort_requested():
```

Comment:
one small suggestion, once the goal is to have this monitor enabled only for youtube videos you can try to check if the playing file starts with "plugin://plugin.video.youtube....". abort_requested makes use of properties and if for some reason this call fails you're using this logic to any file you play through kodi.

Reply:
fixed, getPlayingFile is now checked instead of using a property